x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/results/SparseEmbeddingResults.java

```diff
@@ -178,6 +178,32 @@ public String toString() {
         public Chunk toChunk(ChunkedInference.TextOffset offset) {
             return new Chunk(tokens, offset);
         }
+
+        @Override
+        public Embedding merge(Embedding embedding) {
+            List<WeightedToken> mergedTokens = new ArrayList<>();
+            Set<String> seenTokens = new HashSet<>();
+            int i = 0;
+            int j = 0;
+            // TODO: maybe truncate tokens here when it's getting too large?
+            while (i < tokens().size() || j < embedding.tokens().size()) {
```

Comment:
This requires the token lists to be ordered by weight, maybe just add a note to that effect for the casual reader

```suggestion
            // token lists must be ordered by token weight to be merged correctly
            while (i < tokens().size() || j < embedding.tokens().size()) {
```

Reply:
Added